Source/gendung.cpp

```diff
@@ -149,10 +149,10 @@ void MakeSpeedCels()
 		for (x = 0; x < MAXDUNX; x++) {
 			for (i = 0; i < blocks; i++) {
 				pMap = &dpiece_defs_map_2[x][y];
-				mt = pMap->mt[i];
+				mt = SDL_SwapLE32(pMap->mt[i]);
 				if (mt) {
-					level_frame_count[pMap->mt[i] & 0xFFF]++;
-					level_frame_types[pMap->mt[i] & 0xFFF] = mt & 0x7000;
+					level_frame_count[SDL_SwapLE32(pMap->mt[i] & 0xFFF)]++;

```

Comment:
For the Mac port I'm not seeing any swaps in `MakeSpeedCels`, but I do see additional swaps in `SetDungeonMicros`. Is there a benefit in doing it differently?

Reply:
Hmm maybe the extra swaps in `SetDungeonMicros` are actually because the for loop was split in two (on for when `blocks` are `10` and one for when it's `16`).

Reply:
Looking further in to it it probably is because `SpeedCels` where not used in the Mac render. So this makes your work more compelat then what Blizzard originally did :D

Reply:
Ok I figured out why the game was crashing when I entered the dungeon, all the added swaps in `MakeSpeedCels` should be removed and there is one swap missing in `render.cpp `(see other comment)